changelog/348.improvement.rst

```diff
@@ -0,0 +1 @@
+Add the http headers to the existing domain object so that OpenTelemetry context information can be passed to the action server.
```

Comment:
* how is the header used? That would be interesting to understand for me as I think there could be other options to the necessary information.
* And do we need all the headers?

Reply:
- The header is needed for the trace context info. The w3c standard description is [here](https://www.w3.org/TR/trace-context/#trace-context-http-headers-format).
- We could pass only that w3c trace context but would also be good to pass the [opentracing header](https://www.jaegertracing.io/docs/1.21/client-libraries/#key) since we're in a transition period.

Reply:
Thanks for the explanation!